torch/csrc/distributed/c10d/comm.cpp

```diff
@@ -98,5 +98,27 @@ std::vector<at::Tensor> GradBucket::getGradients() const {
   }
   return per_parameter_tensors;
 }
+namespace detail {
+
+at::Tensor parseCppCommHookResult(const c10::IValue& result)
+{
+  if (result.isPyObject()) {
```

Comment:
Curious, why do we need this change to DDP comm hooks?

Reply:
In the case of a python side future and if the result is an array of tensors, the value will be a python list.

Let me add a test specific for this case to highlight the issue.